src/Common/ProfileEvents.cpp

```diff
@@ -68,6 +68,9 @@
     M(DelayedInserts, "Number of times the INSERT of a block to a MergeTree table was throttled due to high number of active data parts for partition.") \
     M(RejectedInserts, "Number of times the INSERT of a block to a MergeTree table was rejected with 'Too many parts' exception due to high number of active data parts for partition.") \
     M(DelayedInsertsMilliseconds, "Total number of milliseconds spent while the INSERT of a block to a MergeTree table was throttled due to high number of active data parts for partition.") \
+    M(DistributedDelayedInserts, "Number of times the INSERT of a block to a Distributed table was throttled due to high number of pending bytes.") \
+    M(DistributedRejectedInserts, "Number of times the INSERT of a block to a Distributed table was rejected with 'Too many bytes' exception due to high number of pending bytes.") \
+    M(DistributedDelayedInsertsMilliseconds, "Total number of milliseconds spent while the INSERT of a block to a Distributed table was throttled due to high number of pending bytes.") \
```

Comment:
Great. Maybe also overall size of data collected in async metrics?

Reply:
It is still not 100% accurate, but after this PR it became more accurate (since on each INSERT the counters are updated, while before it was updated only during distributed sends or select from system.distribution_queue).

But yes, maybe it still worth adding separate metric for this (even if there is system.distribution_queue)